pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -1119,14 +1119,31 @@ var startTime20190107 = metav1.Date(2019, 1, 7, 1, 1, 1, 0, time.UTC)
 // that podsFitsOnNode works correctly and is tested separately.
 func TestSelectNodesForPreemption(t *testing.T) {
 	defer algorithmpredicates.SetPredicatesOrderingDuringTest(order)()
+
+	filterPluginRegistry := framework.Registry{filterPlugin.Name(): NewFilterPlugin}
```

Comment:
we faced problems in the past when using shared variables for the plugin instances, it would be great if we can change the way we instantiate plugins to be similar to what we did in integration tests:

define a generic function for instantiation:
```
// newPlugin returns a plugin factory with specified Plugin.
func newPlugin(plugin framework.Plugin) framework.PluginFactory {
	return func(_ *runtime.Unknown, fh framework.FrameworkHandle) (framework.Plugin, error) {
		return plugin, nil
	}
}
```

and here, you can just do:
```
filterPlugin := &FilterPlugin{}
	registry := framework.Registry{filterPluginName: newPlugin(filterPlugin)}
```

The above should allow you to remove the global instance ```filterPlugin```

Reply:
DONE